src/constants/action-types.js

```diff
@@ -54,3 +54,17 @@ export const SET_PROGRESS_PRECENTAGE = 'SET_PROGRESS_PRECENTAGE'
 export const SET_ARTICLE_TOOLS = 'SET_ARTICLE_TOOLS'
 
 export const SET_NEXT_POPUP_TIME_STAMP = 'SET_NEXT_POPUP_TIME_STAMP'
+
+// actions on bookmarks
+export const FETCH_MULTIPLE_BOOKMARKS_REQUEST = 'FETCH_MULTIPLE_BOOKMARKS_REQUEST'
+export const FETCH_MULTIPLE_BOOKMARKS_SUCCESS = 'FETCH_MULTIPLE_BOOKMARKS_SUCCESS'
+export const FETCH_MULTIPLE_BOOKMARKS_FAILURE = 'FETCH_MULTIPLE_BOOKMARKS_FAILURE'
+export const FETCH_SINGLE_BOOKMARK_REQUEST = 'FETCH_SINGLE_BOOKMARK_REQUEST'
+export const FETCH_SINGLE_BOOKMARK_SUCCESS = 'FETCH_SINGLE_BOOKMARK_SUCCESS'
+export const FETCH_SINGLE_BOOKMARK_FAILURE = 'FETCH_SINGLE_BOOKMARK_FAILURE'
+export const DELETE_SINGLE_BOOKMARK_REQUEST = 'DELETE_SINGLE_BOOKMARK_REQUEST'
+export const DELETE_SINGLE_BOOKMARK_SUCCESS = 'DELETE_SINGLE_BOOKMARK_SUCCESS'
+export const DELETE_SINGLE_BOOKMARK_FAILURE = 'DELETE_SINGLE_BOOKMARK_FAILURE'
+export const CREATE_SINGLE_BOOKMARK_REQUEST = 'CREATE_SINGLE_BOOKMARK_REQUEST'
+export const CREATE_SINGLE_BOOKMARK_SUCCESS = 'CREATE_SINGLE_BOOKMARK_SUCCESS'
+export const CREATE_SINGLE_BOOKMARK_FAILURE = 'CREATE_SINGLE_BOOKMARK_FAILURE'
```

Comment:
Could we use camelCase here?
Although the old action type conventions are ALL UPPER CASES,
we could have a new convention since this PR.

Reply:
I try to replace the ALL_UPPER_CASES style to camelCase. But I found that ALL_UPPER_CASES has better readability for action types because the request, failure, and success of a given async action looks so similar.

```js
// actions on bookmarks
export const fetchMultipleBookmarksRequest = 'FETCH_MULTIPLE_BOOKMARKS_REQUEST'
export const fetchMultipleBookmarksSuccess = 'FETCH_MULTIPLE_BOOKMARKS_SUCCESS'
export const fetchMultipleBookmarksFAILURE = 'FETCH_MULTIPLE_BOOKMARKS_FAILURE'
export const fetchSingleBookmarkRequest = 'FETCH_SINGLE_BOOKMARK_REQUEST'
export const fetchSingleBookmarkSuccess = 'FETCH_SINGLE_BOOKMARK_SUCCESS'
export const fetchSingleBookmarkFailure = 'FETCH_SINGLE_BOOKMARK_FAILURE'
export const deleteSingleBookmarkRequest = 'DELETE_SINGLE_BOOKMARK_REQUEST'
export const deleteSingleBookmarkSuccess = 'DELETE_SINGLE_BOOKMARK_SUCCESS'
export const deleteSingleBookmarkFailure = 'DELETE_SINGLE_BOOKMARK_FAILURE'
export const createSingleBookmarkRequest = 'CREATE_SINGLE_BOOKMARK_REQUEST'
export const createSingleBookmarkSuccess = 'CREATE_SINGLE_BOOKMARK_SUCCESS'
export const createSingleBookmarkFailure = 'CREATE_SINGLE_BOOKMARK_FAILURE'
```

Reply:
If we use nested objects to represent them?
such  as 
```
// constants/action-types.js
export const singleBookmark = {
    read: {
      request: 'request to fetch single bookmark',
      success: 'succes to fetch single bookmark',
      failure: 'fail to fetch single bookmark'
    },
   create: {
      request: 'request to  create single bookmark',
      success: 'succes to create single bookmark',
      failure: 'fail to create single bookmark'
   },
   delete: {
      request: 'request to  delete single bookmark',
      success: 'succes to delete single bookmark',
      failure: 'fail to delete single bookmark'
   }
}

// actions/bookmarks.js
import { singleBookmark as singleBookmarkAT }  from '../constants/action-types'
store. dispatch({
  actionType: singleBookmarkAT.read.request,
})

```

How do you think?
Will it be easier to understand?

Reply:
Nice